### PR TITLE
Task/ctaa 1374 clean up

### PR DESCRIPTION
**Pull request by dusanjencik**
## Changes

- Removed all code related to P2P.
- Removed all code related to automatic handshakes.
- Removed all code related to manual handshakes.
- Added DB migration to remove all data related to manual and automatic handshake.
- Added preferences migration to remove invalid keys.

## Mergable after
- https://github.com/austrianredcross/stopp-corona-android/pull/85

## Solved issues

- [Issue #1374](https://tasks.pxp-x.com/browse/CTAA-1374)
- [Issue #1480](https://tasks.pxp-x.com/browse/CTAA-1480)

### CODE REVIEW

**Comment by TuxCoder** [comment]
you forgot to remove the maven repo and the api key:

https://github.com/austrianredcross/stopp-corona-android/blob/task/CTAA-1374_clean_up/build.gradle#L26

https://github.com/austrianredcross/stopp-corona-android/blob/task/CTAA-1374_clean_up/app/build.gradle#L63

https://github.com/austrianredcross/stopp-corona-android/blob/task/CTAA-1374_clean_up/app/build.gradle#L92

you can also do `grep p2p -ri <project>` or similar file search tool:

```
./build.gradle
./app/build.gradle
./app/src/main/AndroidManifest.xml
./app/src/main/assets/privacy_en.html
./app/src/main/assets/privacy_de.html
./app/src/main/assets/privacy-and-terms-of-use_en.html
./app/src/main/assets/privacy-and-terms-of-use_de.html

**Comment by dusanjencik** [comment]
@TuxCoder yes, thank you very much for spotting it 👍 I'll fix it on Tuesday.

**Comment by falkorichter** [comment]
@TuxCoder great comment. Text are beeing changed by the colleagues in the legal department as we speak, permissions will all have to be reconsidered. We have decided to separate these two issues. I moved the Bluetooth permission from the block of p2p permissions and we have a new ticket for reconsideration of all permissions. We might need/use some p2p permissions and want to avoid side affects here...

https://tasks.pxp-x.com/browse/CTAA-1547 is the task :) But you might not have access to it 😢 

🤗  👍  ✋  for your contribution!

File: app/src/main/java/at/roteskreuz/stopcorona/constants/Constants.kt [hunk 1]
```diff
@@ -77,7 +77,6 @@ object Constants {
         const val DATA_PRIVACY_REPOSITORY_PREFIX = PREFIX + "terms_and_conditions_repository_"
         const val DASHBOARD_PREFIX = PREFIX + "dashboard_"
         const val QUARANTINE_REPOSITORY_PREFIX = PREFIX + "quarantine_repository_"
-        const val CORONA_DETECTION_REPOSITORY_PREFIX = PREFIX + "corona_detection_repository_"
```
Comment (falkorichter) [inline app/src/main/java/at/roteskreuz/stopcorona/constants/Constants.kt:80]: Note to self, are we clearing the value here?

Comment (falkorichter) [reply app/src/main/java/at/roteskreuz/stopcorona/constants/Constants.kt:80]: Preference is safe to delete, no data is leaked we can just leave it there... @uluckas or should we be extra safe and delete this old value from system preferences?

**Comment by TuxCoder** [comment]
@falkorichter  Thanks  for taking care so fast!

Yeah have sadly no access to the internal issue tracker, that makes contribution currently a bit hard....

File: app/src/main/java/at/roteskreuz/stopcorona/model/repositories/DashboardRepository.kt [hunk 2]
```diff
@@ -2,48 +2,26 @@ package at.roteskreuz.stopcorona.model.repositories
...
-    val showMicrophoneExplanationDialog: Boolean
-
-    /**
-     * Observes the number of met people.
-     */
-    fun observeSavedEncountersNumber(): Observable<Int>
-
     /**
-     * Do not show explanation dialog again.
+     * Information if the automatic handshake was enabled automatically on the first start
```
Comment (falkorichter) [inline app/src/main/java/at/roteskreuz/stopcorona/model/repositories/DashboardRepository.kt:13]: I don´t get this, we don´t have a service anymore...

File: app/src/main/java/at/roteskreuz/stopcorona/model/repositories/DashboardRepository.kt [hunk 3]
```diff
@@ -2,48 +2,26 @@ package at.roteskreuz.stopcorona.model.repositories
...
-        private set
-
-    override fun observeSavedEncountersNumber(): Observable<Int> {
-        return nearbyRecordDao.observeNumberOfRecords().asDbObservable()
-    }
-
-    override fun setMicrophoneExplanationDialogShown() {
-        showMicrophoneExplanationDialog = false
-    }
+    override var serviceEnabledOnFirstStart: Boolean by preferences.booleanSharedPreferencesProperty(PREF_SERVICE_ENABLED_ON_FIRST_START, false)
```
Comment (falkorichter) [inline app/src/main/java/at/roteskreuz/stopcorona/model/repositories/DashboardRepository.kt]: this should all be removed IMHO

File: app/src/main/java/at/roteskreuz/stopcorona/model/repositories/ExposureNotificationRepository.kt [hunk 4]
```diff
@@ -148,7 +147,7 @@ class ExposureNotificationRepositoryImpl(
             return
         }
         registeringWithFrameworkState.loading()
-        exposureNotificationClient.start()
+        exposureNotificationClient.stop()
```
Comment (falkorichter) [inline app/src/main/java/at/roteskreuz/stopcorona/model/repositories/ExposureNotificationRepository.kt:150]: This should have been a fix to the develop branch and not mixed in in here

File: app/src/main/java/at/roteskreuz/stopcorona/model/repositories/ReportingRepository.kt [hunk 5]
```diff
@@ -189,12 +187,13 @@ class ReportingRepositoryImpl(
                 infectionMessengerRepository.storeSentInfectionMessages(resetMessages)
             }
 
-            infectionMessages.addAll(nearbyRecordDao.observeRecordsRecentThan(thresholdTime)
-                .blockingFirst()
-                .map { nearbyRecord ->
-                    nearbyRecord.publicKey to InfectionMessageContent(infectionLevel, nearbyRecord.timestamp)
-                }
-            )
+            // TODO: 28/05/2020 dusanjencik: How to integrate exposure notifications to infection messages?
```
Comment (falkorichter) [inline app/src/main/java/at/roteskreuz/stopcorona/model/repositories/ReportingRepository.kt:191]: https://wiki.sinnerschrader.com/display/SWP/Best+practices#UNUSED

File: app/src/main/java/at/roteskreuz/stopcorona/screens/dashboard/DashboardFragment.kt [hunk 6]
```diff
@@ -202,11 +183,10 @@ class DashboardFragment : BaseFragment(R.layout.fragment_dashboard) {
         /**
          * If the user starts the app for the first time the service will be started automatically
          */
-        // TODO: 27/05/2020 dusanjencik: Decide if we want to enable it automatically or not
-//        if (viewModel.wasServiceEnabledAutomaticallyOnFirstStart.not()) {
-//            viewModel.wasServiceEnabledAutomaticallyOnFirstStart = true
-//            checkDependenciesAndStartAutomaticHandshake(true)
-//        }
+        if (viewModel.wasServiceEnabledAutomaticallyOnFirstStart.not()) {
```
Comment (falkorichter) [inline app/src/main/java/at/roteskreuz/stopcorona/screens/dashboard/DashboardFragment.kt]: wasServiceEnabledAutomaticallyOnFirstStart needs to be renamed. We don´t have no service anymore

Also: I´m unsure if this is a valid operation, unsure if UX approved to register the app automatically for Exposure notifications.

File: app/src/main/java/at/roteskreuz/stopcorona/screens/dashboard/DashboardFragment.kt [hunk 7]
```diff
@@ -231,16 +211,11 @@ class DashboardFragment : BaseFragment(R.layout.fragment_dashboard) {
         }
     }
 
-    // TODO: 27/05/2020 dusanjencik: Remove
-    private fun checkPlayServicesAvailabilityAndStartHandshakeFragment() {
+    private fun checkPlayServicesAvailabilityAndRegisterToExposureNotificationFramework(enableFramework: Boolean) {
```
Comment (falkorichter) [inline app/src/main/java/at/roteskreuz/stopcorona/screens/dashboard/DashboardFragment.kt:214]: I believe this should be handled in another Ticket: https://tasks.pxp-x.com/browse/CTAA-1363 and https://tasks.pxp-x.com/browse/CTAA-1395

File: app/src/main/java/at/roteskreuz/stopcorona/screens/dashboard/DashboardFragment.kt [hunk 8]
```diff
@@ -231,16 +211,11 @@ class DashboardFragment : BaseFragment(R.layout.fragment_dashboard) {
...
         when (GoogleApiAvailability.getInstance().isGooglePlayServicesAvailable(requireContext())) {
             ConnectionResult.SUCCESS -> {
-                if (viewModel.showMicrophoneExplanationDialog) {
-                    // will call startHandshakeFragment() if OK
-                    MicrophoneExplanationDialog().showForResult(REQUEST_MICROPHONE_DIALOG)
-                } else {
-                    startHandshakeFragment()
-                }
+                // TODO: 28/05/2020 dusanjencik: We should check also correct version
+                viewModel.onAutomaticHandshakeEnabled(enableFramework)
```
Comment (falkorichter) [inline app/src/main/java/at/roteskreuz/stopcorona/screens/dashboard/DashboardFragment.kt]: See above

File: app/src/main/java/at/roteskreuz/stopcorona/screens/dashboard/DashboardViewModel.kt [hunk 9]
```diff
@@ -29,21 +29,11 @@ class DashboardViewModel(
...
-//        set(value) {
-//            coronaDetectionRepository.serviceEnabledOnFirstStart = value
-//        }
-
-    var batteryOptimizationDialogShown = false
-
-    fun observeSavedEncounters(): Observable<Int> {
-        return dashboardRepository.observeSavedEncountersNumber()
-    }
+    var wasServiceEnabledAutomaticallyOnFirstStart: Boolean
```
Comment (falkorichter) [inline app/src/main/java/at/roteskreuz/stopcorona/screens/dashboard/DashboardViewModel.kt]: there is no Service anymore

**Review by falkorichter — CHANGES_REQUESTED** [review]

File: app/src/main/java/at/roteskreuz/stopcorona/constants/Constants.kt (hunk 1 above)
Comment (dusanjencik) [reply app/src/main/java/at/roteskreuz/stopcorona/constants/Constants.kt:80]: @falkorichter Already removed in https://github.com/austrianredcross/stopp-corona-android/pull/83/files#diff-4d7fe52c6e225b440d9b5efb6841d6a5R33

File: app/src/main/java/at/roteskreuz/stopcorona/model/repositories/DashboardRepository.kt (hunk 2 above)
Comment (dusanjencik) [reply app/src/main/java/at/roteskreuz/stopcorona/model/repositories/DashboardRepository.kt:13]: Renamed in new commit

File: app/src/main/java/at/roteskreuz/stopcorona/model/repositories/DashboardRepository.kt (hunk 3 above)
Comment (dusanjencik) [reply app/src/main/java/at/roteskreuz/stopcorona/model/repositories/DashboardRepository.kt]: not removed, renamed. https://github.com/austrianredcross/stopp-corona-android/pull/83#discussion_r433731286

File: app/src/main/java/at/roteskreuz/stopcorona/model/repositories/ExposureNotificationRepository.kt (hunk 4 above)
Comment (dusanjencik) [reply app/src/main/java/at/roteskreuz/stopcorona/model/repositories/ExposureNotificationRepository.kt:150]: Moved to https://github.com/austrianredcross/stopp-corona-android/pull/85

Comment (dusanjencik) [reply app/src/main/java/at/roteskreuz/stopcorona/model/repositories/ExposureNotificationRepository.kt:150]: Updated PR description

File: app/src/main/java/at/roteskreuz/stopcorona/model/repositories/ReportingRepository.kt (hunk 5 above)
Comment (dusanjencik) [reply app/src/main/java/at/roteskreuz/stopcorona/model/repositories/ReportingRepository.kt:191]: If there is todo comment, some code might be commented. Will be fixed in later PR.

File: app/src/main/java/at/roteskreuz/stopcorona/screens/dashboard/DashboardFragment.kt (hunk 6 above)
Comment (dusanjencik) [reply app/src/main/java/at/roteskreuz/stopcorona/screens/dashboard/DashboardFragment.kt]: Renamed https://github.com/austrianredcross/stopp-corona-android/pull/83#discussion_r433731286

File: app/src/main/java/at/roteskreuz/stopcorona/screens/dashboard/DashboardFragment.kt (hunk 7 above)
Comment (dusanjencik) [reply app/src/main/java/at/roteskreuz/stopcorona/screens/dashboard/DashboardFragment.kt:214]: Agree. In the code, there is already a todo comment for it: `// TODO: 28/05/2020 dusanjencik: We should check also correct version`

File: app/src/main/java/at/roteskreuz/stopcorona/screens/dashboard/DashboardFragment.kt (hunk 8 above)
Comment (dusanjencik) [reply app/src/main/java/at/roteskreuz/stopcorona/screens/dashboard/DashboardFragment.kt]: Renamed

File: app/src/main/java/at/roteskreuz/stopcorona/screens/dashboard/DashboardViewModel.kt (hunk 9 above)
Comment (dusanjencik) [reply app/src/main/java/at/roteskreuz/stopcorona/screens/dashboard/DashboardViewModel.kt]: Done in https://github.com/austrianredcross/stopp-corona-android/pull/83#discussion_r433731286

File: app/src/main/java/at/roteskreuz/stopcorona/model/repositories/DashboardRepository.kt (hunk 3 above)
Comment (falkorichter) [reply app/src/main/java/at/roteskreuz/stopcorona/model/repositories/DashboardRepository.kt]: yes - renamed!

**Review by falkorichter — APPROVED** [review]

**Pull request merged**